napari/_qt/qt_main_window.py

```diff
@@ -1729,14 +1731,93 @@ def export_figure(
             imsave(path, img)
         return img
 
+    def export_rois(
+        self,
+        rois: list[np.ndarray],
+        paths: Optional[Union[str, Path, list[Union[str, Path]]]] = None,
+        scale: Optional[float] = None,
+    ):
+        """Export the shapes rois with storage file paths.
+
+        For each shape, moves the camera to the center of the shape
+        and adjust the canvas size to fit the shape.
+        Note: The shape height and width can be of type float.
+        However, the canvas size only accepts a tuple of integers.
+        This can result in slight misalignment.
+
+        Parameters
+        ----------
+        rois: list[np.ndarray]
+            A list of arrays  with each being of shape (4, 2) representing a rectangular roi.
+        paths: str, Path, list[str, Path], optional
+            Where to save the rois. If a string or a Path, a directory will be created if it does not exist yet and
+            screenshots will be saved with filename `roi_{n}.png` where n is the nth roi. If paths is a list of either
+            string or paths, these need to be the full paths of where to store each individual roi. In this case
+            the length of the list and the number of rois must match.
+
+        Returns
+        -------
+        screenshot_list: list
+            The list with roi screenshots.
+
+        """
+        if (
+            paths is not None
+            and isinstance(paths, list)
+            and len(paths) != len(rois)
+        ):
+            raise ValueError(
+                trans._(
+                    'The number of file paths does not match the number of ROI shapes',
+                    deferred=True,
+                )
+            )
+
+        if isinstance(paths, (str, Path)):
+            storage_dir = Path(paths)
+            storage_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
There is some weirdness going on here. If I write  `viewer.export_rois(rois, paths='~/napari-stuff/rois')`, then this line makes the directories `~/napari-stuff/rois` in my local directory (not intended, but ok). However, later when processing the call `self.screenshot(path=path, canvas_only=True, scale=scale)`, the path is at some stage unpacked to `/home/my-user/napari-stuff/rois`, and we end up with an error `FileNotFoundError: The directory '/home/my-user/napari-stuff/rois' does not exist`.

I think either the `~` should be unpacked to the absolute path from the beginning, and the directory `/napari-stuff/rois` made in the home folder, **or** the screenshots should just be saved to the `~/napari-stuff/rois` folder created in the current working directory.